nixos/modules/security/pam.nix

```diff
@@ -258,7 +268,7 @@ let
             auth [default=die success=done] ${pam_ccreds}/lib/security/pam_ccreds.so action=validate use_first_pass
             auth sufficient ${pam_ccreds}/lib/security/pam_ccreds.so action=store use_first_pass
           ''}
-          ${optionalString (!(config.security.pam.enableEcryptfs || cfg.pamMount)) "auth required pam_deny.so"}
```

Comment:
Yes, this line is problematic. Given the way the NixOS pam stack is set up, the expectation is that the `auth required pam_deny.so` must always be at the bottom.